lib/project_types/script/layers/infrastructure/languages/task_runner.rb

```diff
@@ -15,14 +17,45 @@ def self.for(ctx, language, script_name)
           end
 
           def check_system_dependencies!
+            required_tool_versions.each { |tool| check_tool_version!(tool[:tool_name], tool[:min_version]) }
+          end
+
+          def install_dependencies
+            check_system_dependencies!
+            output, status = ctx.capture2e(self.class::INSTALL_COMMAND)
+            raise Errors::DependencyInstallationError, output unless status.success?
```

Comment:
Do you need the standard output in the raised error? If not, I'd recommend using `capture3` and only including the content piped through standard error:

```rb
stdout, stderr, status = ctx.capture3(self.class::INSTALL_COMMAND)
raise Errors::DependencyInstallationError, stderr unless status.success?
```
Do we expect that error to be handled elsewhere? If not, I'd raise `ShopifyCLI::Bug` instead. Once we get more data around when this scenario might happen, we can add some logic it to gracefully handle those scenarios.

Reply:
In that case you can raise `ShopifyCLI::Abort`. These are the available options and how to use them (we'll document them soon):
- [`ShopifyCLI::Bug`](https://github.com/Shopify/shopify-cli/blob/main/lib/shopify_cli.rb#L72): It's a bug and needs to be reported to Bugsnag. It's important that you attach metadata that helps us better understand why it happens and add some logic to handle the scenario gracefully or prevent it from happening.
- [`ShopifyCLI::BugSilent`](https://github.com/Shopify/shopify-cli/blob/main/lib/shopify_cli.rb#L73): Same as above but the error doesn't get printed. You usually use this one when you want to control how the error is printed: `puts "my custom error"; raise ShopifyCLI::BugSilent, "error"`
-  [`ShopifyCLI::Abort`](https://github.com/Shopify/shopify-cli/blob/main/lib/shopify_cli.rb#L71): This one terminates the execution and doesn't report anything to Bugsnag. 
- [`ShopifyCLI::AbortSilent`](https://github.com/Shopify/shopify-cli/blob/main/lib/shopify_cli.rb#L74): Same as `Abort` but without printing anything.

In your case, because it's a user error and we can't do anything about that you should raise a `ShopifyCLI::Abort` with an actionable message that tells the user why it happened and what they can do about that.